doc/features/vectors/README.md

```diff
@@ -0,0 +1,148 @@
+
+```csharp
+var vector1 = new CqlVector<int>(1, 2, 3);
+var vector2 = new CqlVector<int>(1, 2, 3);
+vector1.Equals(vector2); // this returns true
+```
+
+## Writing vector data and performing vector search operations
+
+The `vector` type is handled by the driver the same way as any other CQL type.
+
+The following examples use this schema. In this case, `j` is a 3 dimensional `vector` column of `float` values. Both the vector subtype and the number of dimensions can be changed. Any CQL type is valid as a vector subtype.
+
+```sql
+CREATE TABLE IF NOT EXISTS table1 (
+     i int PRIMARY KEY, 
+     j vector<float, 3>
+);
+
+/* Supported by C* 5.0, for vector search with the ANN operator */
+CREATE CUSTOM INDEX IF NOT EXISTS ann_table1_index ON table1(j) USING 'StorageAttachedIndex';
+```
+
+### Simple Statements
+
+```csharp
+await session.ExecuteAsync(
+    new SimpleStatement(
+        "INSERT INTO table1 (i, j) VALUES (?, ?)", 
+        1, 
+        new CqlVector<float>(1.0f, 2.0f, 3.0f)));
+var rowSet = await session.ExecuteAsync(
+    new SimpleStatement(
+        "SELECT * FROM table1 ORDER BY j ANN OF ? LIMIT ?", 
+        new CqlVector<float>(0.6f, 0.5f, 0.9f),
+        1));
+var row = rowSet.Single();
+var i = row.GetValue<int>("i");
+var j = row.GetValue<CqlVector<float>?>("j");
+```
+
+### Prepared Statements
+
+```csharp
+var psInsert = await session.PrepareAsync("INSERT INTO table1 (i, j) VALUES (?, ?)");
+var psSelect = await session.PrepareAsync("SELECT * FROM table1 ORDER BY j ANN OF ? LIMIT ?");
+
+var boundInsert = psInsert.Bind(2, new CqlVector<float>(5.0f, 6.0f, 7.0f));
+await session.ExecuteAsync(boundInsert);
+
+var boundSelect = psSelect.Bind(new CqlVector<float>(4.7f, 5.0f, 5.0f), 1);
+var rowSet = await session.ExecuteAsync(boundSelect);
+
+var row = rowSet.Single();
+var i = row.GetValue<int>("i");
+var j = row.GetValue<CqlVector<float>>("j");
+```
+
+### LINQ and Mapper
+
+The LINQ component of the driver doesn't support the `ANN` operator so it's probably best to avoid using LINQ when working with vectors. If a particular workload doesn't require the `ANN` operator then LINQ can be used without issues.
+
+```csharp
+// you can also provide a MappingConfiguration object to the Table/Mapper constructors 
+// (or use MappingConfiguration.Global) programatically instead of these attributes
+[Cassandra.Mapping.Attributes.Table("table1")]
+public class Table1
+{
+    [Cassandra.Mapping.Attributes.PartitionKey]
+    [Cassandra.Mapping.Attributes.Column("i")]
+    public int I { get; set; }
+
+    [Cassandra.Mapping.Attributes.Column("j")]
+    public CqlVector<float>? J { get; set; }
+}
+
+// LINQ
+
+var table = new Table<TestTable1>(session);
+await table
+    .Insert(new TestTable1 { I = 3, J = new CqlVector<float>(10.1f, 10.2f, 10.3f) })
+    .ExecuteAsync();
+
+// DON'T USE AllowFiltering, this is required in this case because the ANN operator 
```

Comment:
I couldn't understand this. Why I shouldn't use `AllowFiltering`?

Reply:
It performs a full scan on all C* nodes so the performance can be very unpredictable. [Here's some docs](https://cassandra.apache.org/doc/stable/cassandra/cql/dml.html#allow-filtering) on this topic from the official C* docs

Reply:
Got it. Do you think we can consider the following wording, and adding another way of LINQ?
```csharp
// Using AllowFiltering is not recommended due to unpredictable performance.
// Here we use AllowFiltering in this case just for the example,
// as the ANN operator is not supported in LINQ yet.
var entity = (await table.Where(t => t.I == 3 && t.J == CqlVector<float>.New(new [] {10.1f, 10.2f, 10.3f})).AllowFiltering().ExecuteAsync()).SingleOrDefault();
// The following way also works
var entity = (await (from t in table where t.J == CqlVector<float>.New(new [] {10.1f, 10.2f, 10.3f}) select t).AllowFiltering().ExecuteAsync()).SingleOrDefault();
```

Reply:
updated this section, can you give it another look?